app/src/main/java/com/example/androidacademyhomework/UI/MovieDetailsFragment.kt

```diff
@@ -6,11 +6,32 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.Button
+import android.widget.RatingBar
+import android.widget.TextView
 import androidx.navigation.fragment.findNavController
+import androidx.navigation.fragment.navArgs
+import androidx.recyclerview.widget.DefaultItemAnimator
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
 import com.example.androidacademyhomework.R
+import com.example.androidacademyhomework.adapters.ActorsAdapter
+import com.example.androidacademyhomework.domain.MoviesDataSource
+import com.example.androidacademyhomework.models.Movie
 
 class MovieDetailsFragment : Fragment() {
 
+    private lateinit var textViewDetailFragmentAgeLimit: TextView
```

Comment:
You should read about fragment lifecycle and his view lifecycle. By this, you're making a memory leak!

Reply:
Fixed